pkgs/by-name/at/atf/package.nix

```diff
@@ -6,7 +6,13 @@
   autoreconfHook,
   kyua,
   gitUpdater,
-}:
+  darwin,
+}@args:
+
+let
+  # atf is a dependency of libiconv. Avoid an infinite recursion with `pkgsStatic` by using a bootstrap stdenv.
+  stdenv = if args.stdenv.hostPlatform.isDarwin then darwin.bootstrapStdenv else args.stdenv;
```

Comment:
This means that it’s impossible to override the `stdenv` on Darwin (without knowing implementation details of the derivation). Is there a way to do this that preserves the ability to override it? That’s why I originally did it in `all-packages.nix`.

Reply:
I think @emilazy's comment in https://github.com/NixOS/nixpkgs/pull/453948#discussion_r2445872583 applies to that?

Reply:
It can’t be in the bootstrap because then `pkgsStatic` will have an infinite recursion. To make things behave the same between native and cross, the “overrides” have to be in the package definitions not in the bootstrap overlay.

Reply:
Putting as draft for the moment. I am currently looking into a way to make this case better to solve here - it's really unsatisfactory, that we can't override `stdenv` for these packages afterwards anymore.